compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -753,6 +755,7 @@ impl<'a, 'tcx> FindInferSourceVisitor<'a, 'tcx> {
                     GenericArgKind::Lifetime(_) => 0, // erased
                     GenericArgKind::Type(ty) => self.ty_cost(ty),
                     GenericArgKind::Const(_) => 3, // some non-zero value
+                    GenericArgKind::Effect(_) => 3, // some non-zero value
```

Comment:
would put them together with regions, you cannot explicitly mention effects so they don't have a cost